lib/rucio/core/rule.py

```diff
@@ -1943,7 +2263,14 @@ def update_rules_for_lost_replica(scope, name, rse_id, nowait=False, *, session:
         elif rule.locks_replicating_cnt == 0 and rule.locks_stuck_cnt == 0:
             rule.state = RuleState.OK
             if rule.grouping != RuleGrouping.NONE:
-                session.query(models.DatasetLock).filter_by(rule_id=rule.id).update({'state': LockState.OK})
+                stmt = update(
+                    models.DatasetLock
+                ).where(
+                    models.DatasetLock.rule_id == rule.id
+                ).values(
+                    state=LockState.OK
```

Comment:
I'm not sure if this would work - you might need to pass it like this instead: `{'state': LockState.OK}` (I'm assuming this based on the other update statements in this commit)

Reply:
It should work anyway, based on the docs https://docs.sqlalchemy.org/en/20/core/dml.html#sqlalchemy.sql.expression.update

feel free to ignore this comment